app/grandchallenge/reader_studies/models.py

```diff
@@ -347,12 +383,22 @@ def assign_permissions(self):
 
     def clean(self):
         # Make sure that the image port is only set when using drawn
-        # annotations. At the moment 2D bounding boxes are the only drawn type.
-        if (self.answer_type == self.ANSWER_TYPE_2D_BOUNDING_BOX) != bool(
-            self.image_port
-        ):
+        # annotations.
+        if (
+            self.answer_type
+            in [
+                self.ANSWER_TYPE_2D_BOUNDING_BOX,
+                self.ANSWER_TYPE_DISTANCE_MEASUREMENT,
+            ]
+        ) != bool(self.image_port):
+            raise ValidationError(
+                "The image port must (only) be set for annotation questions."
+            )
+
+        if self.answer_type == self.ANSWER_TYPE_BOOL and self.required:
```

Comment:
"Required" for text would mean "something other than whitespace", I guess? Good catch otherwise!

Reply:
Yes, my intention for this is the same as htmls input required attr: "specifies that an input field must be filled out before submitting the form". Will simplify the implementation if we ever generate pure html forms :-)